api/kv/value/value.go

```diff
@@ -183,11 +214,24 @@ func (v Value) AsString() string {
 	return v.stringly
 }
 
+// AsArray array assign interface array value, if given value mataches
+// stored array type.
+func (v Value) AsArray(i interface{}) error {
```

Comment:
I'm not sure this function signature is going to be useful. It requires the caller to have prior knowledge of what the return type should be. This is probably not a great assumption.

Maybe just return the array as an interface and let the caller introspect to handle it?

```suggestion
func (v Value) AsArray() interface{} {
```

I'm interested in what others think.

Reply:
The return type from method was the most tricky part in this task, because I wasn't sure will use type casting value after getting it or we handle it same way `json.Unmarshal` does.

Reply:
+1 to returning the `interface{}` type. Unlike in the `json.Unmarshal` case where the caller is expected to have understanding of what structure and types are being unmarshaled, here the user does not have that info. They will need to interrogate the result. If they are expected to pass the correct type and not able to interrogate what this type they need to pass is they will be unable to use this function.

Reply:
Since no one else gave any suggestion so, I changed function signature to what you have requested.